TRAMS-API/Controllers/ProjectsController.cs

```diff
@@ -0,0 +1,102 @@
+﻿using API.Mapping;
+using API.Models.D365;
+using API.Models.Request;
+using API.Repositories;
+using API.Repositories.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace API.Controllers
+{
+    /// <summary>
+    /// API controller for retrieving Academy Transfers projects from TRAMS
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    [Route("[controller]")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(string), StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
+    public class ProjectsController : Controller
+    {
+        private readonly IProjectsRepository _projectsRepository;
+        private readonly IAcademiesRepository _academiesRepository;
+        private readonly ITrustsRepository _trustsRepository;
+        private readonly IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> _mapper;
+
+        public ProjectsController(IProjectsRepository projectsRepository,
+                                  IAcademiesRepository academiesRepository,
+                                  ITrustsRepository trustsRepository,
+                                  IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> mapper)
+        {
+            _projectsRepository = projectsRepository;
+            _academiesRepository = academiesRepository;
+            _trustsRepository = trustsRepository;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// API endpoint for creating a new Academy Transfers Project
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("/projects/")]
+        [ProducesResponseType(typeof(string), StatusCodes.Status204NoContent)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status422UnprocessableEntity)]
+        public async Task<IActionResult> InsertTrust([FromBody]PostProjectsRequestModel model)
+        {
+            var debug = 0;
+
+            var projectAcademiesIds = model.ProjectAcademies.Select(a => a.AcademyId).ToList();
+
+            foreach(var academyId in projectAcademiesIds)
+            {
+                var academy = await _academiesRepository.GetAcademyById(academyId);
+
+                if (academy == null)
+                {
+                    return new UnprocessableEntityObjectResult($"No academy found with the id of: {academyId}");
```

Comment:
Suggestion: Unsure over the story requirements here but wondering if it's worth checking all academies and then returning an aggregated result so that the consumer knows the full state of play.

Reply:
Will do